src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -340,13 +342,26 @@ public static void DetectInvalidValueForParentProperty(SemanticModel semanticMod
                     if (semanticModel.ResourceMetadata.TryLookup(baseSyntax) is not { } && !semanticModel.GetTypeInfo(baseSyntax).IsError())
                     {
                         // we throw an error diagnostic when the parent property contains a value that cannot be computed or does not directly reference another resource.
-                        // this includes ternary operator expressions, which Bicep does not support 
+                        // this includes ternary operator expressions, which Bicep does not support
                         diagnosticWriter.Write(referenceParentSyntax, x => x.InvalidValueForParentProperty());
                     }
                 }
             }
         }
 
+        private static void BlockLambdasOutsideFunctionArguments(SemanticModel model, IDiagnosticWriter diagnostics)
+        {
+            CallbackVisitor.Visit(model.Root.Syntax, syntax => {
+                if (syntax is LambdaSyntax lambdaSyntax &&
+                    model.Binder.GetParent(lambdaSyntax) is not FunctionArgumentSyntax functionArgument)
```

Comment:
Is there any reasonable occasion when FunctionArgumentSyntax would be an ancestor at a slight remove? The only scenario I could come up with is if the lambda where in a parenthetical expression (e.g., something like `map(arrayArg, (foo => foo.bar))` where the parentheses don't alter the semantics but might be there for readability). Not supporting parenthetical parents doesn't seem like much of a limitation, but I wasn't sure if there were other syntax types that might get interposed between the lambda and the FunctionArgumentSyntax node.

Reply:
Parentheses is the only one I can think of - any other parent syntax will modify the semantics and should be blocked. I did debate special-casing parentheses, but (a) got lazy and (b) thought it's actually simpler to explain "lambdas can only be used in function arguments"

Reply:
Will ARM allow an `if()` expression that resolves to a lambda in place of a lambda? Not sure if a ternary might be able to result in a legal template. Although users could always put the ternary within the lambda (`map(arr, double ? (i => i * 2) : (i => i))` is more succinctly but marginally less efficiently expressed as `map(arr, i => double ? i * 2 : i)`).

Reply:
I agree that people probably won't put the lambdas in parentheses, but seems kind of hacky to block that. I can't think of any other case than parentheses, either.

Reply:
No, ARM won't allow a lambda inside an `if()` - as of today, `if` requires the argument to be evaluated before it can be evaluated into a value, and a lambda function cannot be directly evaluated. This is probably something we could relax in the future if desired, but would require deployment engine changes.

Reply:
> I agree that people probably won't put the lambdas in parentheses, but seems kind of hacky to block that. I can't think of any other case than parentheses, either.

Btw, my CodeFlow was out of sync, so I replied out of order I. (Was talking about the parenthesized lambdas). Makes sense on `if()`.

Reply:
Done